WordPress/gutenberg
# Block API: Support and bootstrap server-registered block attribute schemas

Related: #1905, #886

This pull request is an extension of #1905, allowing blocks registered on the server to define their attributes structure using the same JSON-schema-like syntax supported in the client. Further, blocks registered this way can omit these attributes in their JavaScript registration, instead taking advantage of automatic attributes bootstrapping from the server to the client.

This replaces the need for blocks registered on the server to handle their own validation and defaulting. Attributes defined in a server block registration are automatically validated using [`rest_validate_value_from_schema`](https://developer.wordpress.org/reference/functions/rest_validate_value_from_schema/).

Included is a port of the Latest Posts blocks attributes definition from client to server.

__Before:__

https://github.com/WordPress/gutenberg/blob/b5ef57a16d4610a6aa01ae434cdc1efb2a7fc4ca/blocks/library/latest-posts/index.js#L39-L59

__After:__

https://github.com/WordPress/gutenberg/blob/1851f5a6c11195b1aceec3efbb9714acbaa2cc4d/blocks/library/latest-posts/index.php#L66-L87

__Testing instructions:__

Verify that there are no regressions in the insertion and saving of a Latest Posts block. In fact, note that the client now correctly defaults alignment to center, which was previously applied only in the server rendering.

Ensure Mocha and PHPUnit tests pass:

```
npm test
```

```
phpunit
```
